server/src/main/scala/controllers/Application.scala

```diff
@@ -678,56 +678,39 @@ class Application @Inject()(implicit val config: Configuration,
   }
 
   def exportFlightsWithSplitsBetweenTimeStampsCSV(start: String, end: String, terminalName: TerminalName): Action[AnyContent] = Action.async {
-    val startPit = getLocalNextMidnight(SDate(start.toLong))
+    val startPit = getLocalLastMidnight(SDate(start.toLong))
     val endPit = getLocalNextMidnight(SDate(end.toLong))
 
     val portCode = airportConfig.portCode
     val fileName = f"$portCode-$terminalName-arrivals-" +
       f"${startPit.getFullYear()}-${startPit.getMonth()}%02d-${startPit.getDate()}-to-" +
       f"${endPit.getFullYear()}-${endPit.getMonth()}%02d-${endPit.getDate()}"
 
-    val days: Seq[Future[Option[String]]] = (startPit.millisSinceEpoch to endPit.millisSinceEpoch by oneDayMillis).map(
-      millis => flightsCSVFromCrunchState(terminalName, MilliDate(millis), 0, 24)
-    )
-
-    val csvDataFuture = Future.sequence(days.map(fd => fd.recoverWith {
-      case e =>
-        log.error(s"Failed to recover flights for day ${e.getMessage}")
-        Future(None)
-    }))
-      .map(_.collect {
-        case Some(s) => s
-      }.mkString("\n"))
+    val days: Seq[Future[Option[String]]] = (startPit.millisSinceEpoch to endPit.millisSinceEpoch by oneDayMillis).zipWithIndex.map {
+      case (millis, index) =>
+        flightsCSVFromCrunchState(terminalName, MilliDate(millis), 0, 24, loadBestCrunchStateForPointInTime(millis), index == 0)
```

Comment:
would it be worth using named parameters for this to make it clear what some of them them are?